src/bare-metal/no_std.md

```diff
@@ -0,0 +1,65 @@
+# `no_std`
+
+<table>
+<tr>
+<th>
+  
+`core`
+
+</th>
+<th>
+
+`alloc`
+
+</th>
+<th>
+
+`std`
+
+</th>
+</tr>
+<tr valign="top">
+<td>
+
+* Slices, `&str`, `CStr`
+* `NonZeroU8`...
+* `Option`, `Result`
+* `Display`, `Debug`, `write!`...
+* `Iterator`
+* `panic!`, `assert_eq!`...
+* `NonNull` and all the usual pointer-related functions
+* `Future` and `async`/`await`
+* `fence`, `AtomicBool`, `AtomicPtr`, `AtomicU32`...
+* `Duration`
+
+</td>
+<td>
+
+* `Box`, `Cow`, `Arc`, `Rc`
+* `Vec`, `BinaryHeap`, `BtreeMap`, `LinkedList`, `VecDeque`
+* `String`, `CString`, `format!`
+
+</td>
+<td>
+
+* `Error`
+* `HashMap`
+* `Mutex`, `Condvar`, `Barrier`, `Once`, `RwLock`, `mpsc`
+* `File` and the rest of `fs`
+* `println!`, `Read`, `Write`, `Stdin`, `Stdout` and the rest of `io`
+* `Path`, `OsString`
+* `net`
+* `Command`, `Child`, `ExitCode`
+* `spawn`, `sleep` and the rest of `thread`
+* `SystemTime`, `Instant`
+
+</td>
+</tr>
+</table>
```

Comment:
The hand-written HTML does not look very nice, it bumps into the speaker notes below:

![image](https://user-images.githubusercontent.com/89623/225894964-2553704e-6be1-41e1-8cc9-82868ea19f90.png)

I instead format it as a simple list, possibly split across three pages like I suggested above:

```suggestion

## `core`

* `&[T]`, `&str`, `CStr`
* `NonZeroU8`...
* `Option`, `Result`
* `Display`, `Debug`, `write!`...
* `Iterator`
* `panic!`, `assert_eq!`...
* `NonNull` and pointer-related functions
* `Future` and `async`/`await`
* `fence`, `AtomicBool`, `AtomicPtr`, `AtomicU32`...
* `Duration`

## `alloc`

* `Box`, `Cow`, `Arc`, `Rc`
* `Vec`, `BinaryHeap`, `BtreeMap`, `LinkedList`, `VecDeque`
* `String`, `CString`, `format!`

## `std`

* `Error`
* `HashMap`
* `Mutex`, `Condvar`, `Barrier`, `Once`, `RwLock`, `mpsc`
* `File` and the rest of `fs`
* `println!`, `Read`, `Write`, `Stdin`, `Stdout` and the rest of `io`
* `Path`, `OsString`
* `net`
* `Command`, `Child`, `ExitCode`
* `spawn`, `sleep` and the rest of `thread`
* `SystemTime`, `Instant`
```